core/arch/arm/include/kernel/misc.h

```diff
@@ -28,15 +28,82 @@
 #ifndef KERNEL_MISC_H
 #define KERNEL_MISC_H
 
+#include <mm/mobj.h>
 #include <types_ext.h>
 #include <arm.h>
 #include <kernel/thread.h>
 
+enum memparam_dir {
+	MEMPARAM_DIR_IN = 0,
+	MEMPARAM_DIR_OUT,
+	MEMPARAM_DIR_INOUT,
+};
+
 size_t get_core_pos(void);
 
 uint32_t read_mode_sp(int cpu_mode);
 uint32_t read_mode_lr(int cpu_mode);
 
+/**
```

Comment:
`msg_param` could be used as both prefix and basename of the file.
`extract_pages_from_params()` would then become `msg_param_extract_pages()` and so on.